### PR TITLE
Automated cherry pick of #41771

### PR DESCRIPTION
**Pull request by cblecker**
Cherry pick of #41771 on release-1.5.

#41771: Bump golang versions to 1.7.5

### CODE REVIEW

**Comment by ixdy** [comment]
@k8s-bot ok to test
/lgtm

**Comment by ixdy** [comment]
/lgtm cancel
@k8s-bot cross build this

Hm, one thing I just realized: this effectively cherry-picks part of #38926 in (since the cross-build image changes from that PR are incorporated in the 1.7.5 image), but without the build script changes in #38926. @luxas should we cherry-pick your PR? or will things just work as-is?

**Comment by ixdy** [comment]
fat-fingered the close button, oops

**Comment by ixdy** [comment]
/cancel lgtm
?

**Comment by ixdy** [comment]
@k8s-bot cross build this

**Comment by luxas** [comment]
I see what you mean @ixdy, please hold this for a while yet, I don't wanna break crossbuilds on the v1.5 branch. We should be very careful here

Must go afk in a second... I'll come back to this later

**Comment by mwielgus** [comment]
@k8s-bot test this

**Comment by cblecker** [comment]
I was thinking about this one.. while cherrypicking the changes from master into 1.5 may fix this for this build, would it make sense to incorporate the release version into the image name/tag? It would fix this problem in particular (so minor changes can be made on a per-branch basis without having to have one cross build image for all branches per go version), but it might introduce other issues like a proliferation of images to manage.

**Comment by ixdy** [comment]
This is the first time we've run into an issue like this with the crossbuild image. I think this situation is a little unique and probably isn't worth fixing more generally.

**Comment by mwielgus** [comment]
This pr has broken tests - will not go to 1.5.4 which is about to be cut shortly. Once fixed can go to the next patch release.

**Comment by cblecker** [comment]
SGTM. I believe we're just in a holding pattern until @luxas has some time to look at it.

**Comment by luxas** [comment]
@ixdy I think we should find a way to branch this image in this case, because I don't want to include all the changes I made in https://github.com/kubernetes/kubernetes/pull/38926, that would be too risky IMO.
I don't know if v1.5 and go1.8 works together, but I'm 99% sure it doesn't, since I've seen some hotfixes for being able to run k8s with go1.8 in the v1.6 timeframe. 

So I think go1.8 and v1.6 might work somewhat together, but not v1.5.

@ixdy How do you think we should branch this image? Any preference?

File: build/build-image/cross/VERSION [hunk 1]
```diff
@@ -1 +1 @@
-v1.7.4-0
```
Comment (ixdy) [inline build/build-image/cross/VERSION]: I guess we can modify this tag to include the k8s version in it. maybe v1.7.5-k8s1.5-0?

I should double-check that there aren't any other changes we should cherrypick if we do that.

Comment (cblecker) [inline build/build-image/cross/VERSION]: That would make sense to me!

Comment (ixdy) [inline build/build-image/cross/VERSION]: can you make that change here? I'll make sure everything else looks OK and then can push a new image.

Comment (cblecker) [inline build/build-image/cross/VERSION]: Done!

**Comment by ixdy** [comment]
Sorry, this fell off my radar in the last few days.

I wanted to make sure that we weren't missing any other changes since `v1.7.4-0` of this image.

I found the Dockerfile for that version at https://github.com/kubernetes/kubernetes/blob/dd827035929ebab862ef52756ea9668900f4683f/build-tools/build-image/cross/Dockerfile - it looks like the only diff is the golang version, which is what we expect.

**Comment by ixdy** [comment]
@k8s-bot build this
@k8s-bot test this

**Comment by ixdy** [comment]
@k8s-bot gke test this

**Comment by cblecker** [comment]
@k8s-bot cvm gke e2e test this
@k8s-bot gci gke e2e test this

**Comment by cblecker** [comment]
All builds passed 🎉 !

@mwielgus Over to you for review and cherrypick approval :)

**Comment by cblecker** [comment]
@mwielgus Anything else you need from me to be able to pick this into 1.5? Thank you!

**Comment by ixdy** [comment]
@mwielgus can we merge this in time to catch 1.5.6?

**Comment by mwielgus** [comment]
Ups. I must have missed this completely. How critical is it to get it to 1.5.6?

**Comment by luxas** [comment]
@mwielgus This isn't as critical as cutting v1.5.6 now in order to be able to release v1.6.

This can wait until v1.5.7, although it's unfortunate. We can't risk another (possible) breakage, the changes should be kept at a minimum this week to get a stable v1.6 out

**Comment by cblecker** [comment]
Yeah, I'd agree with @luxas. It would be nice to have in, but definitely shouldn't hold anything up that's needed for the 1.6 release.

**Comment by ixdy** [comment]
@mwielgus also, it didn't have the v1.5 milestone until late last week, which might be why you missed it.

**Pull request merged**